### PR TITLE
Support report_prolog_whitespace

### PR DESCRIPTION
**Pull request by rfscholte**
For the Apache maven-release-plugin we use jdom to rewrite the pom. The code contains patches to preserve prolog whitespaces. With stax-api2 it is possible to detect these, but JDom2 don't support it yet.
This patch will fix that.

### CODE REVIEW

**Comment by rolfl** [comment]
I am taking a moment to run through the implications of allowing Document tontent to contain Text (even just whitespace). This needs to be confirmed against the XML specification, and so on.

Even if the whitespace content in Document is OK, there's still a problem with the whitespace check. It would be better to use the function Verifier.isAllXMLWhiteSpace().

**Comment by rolfl** [comment]
Just so you know, I have pulled your changes, then cherry-picked them in to a new branch. JDOM's current version is not on the master branch (I know, it's confusing). I have pulled things on to the jdom-2.0.x branch, and pushed things up here as new branch prologspace. I have also changed the whitspace handler to use the verifier. I will be using that branch to merge the code through.

File: test/src/java/org/jdom2/test/cases/input/TestStAXStreamBuilder.java [hunk 1]
```diff
@@ -113,16 +116,39 @@ public void testDocTypeSimpleDocument() {
...
 	}
-	
+
+	@Test
+	public void testReportPrologWsDocument() throws Exception {
+		URLClassLoader classLoader = new URLClassLoader(
+				new URL[] { new File("lib/woodstox-core-5.0.2.jar").toURI().toURL(),
+							new File("lib/stax2-api-3.1.4.jar").toURI().toURL() });
+		
+		checkStAX("/DOMBuilder/report_prolog_ws.xml", false, "javax.xml.stream.XMLInputFactory", classLoader, 5);
```
Comment (rfscholte) [inline test/src/java/org/jdom2/test/cases/input/TestStAXStreamBuilder.java]: Woodstox has a provider ( META-INF/services/javax.xml.stream.XMLInputFactory ) to define its implementation of the XmlInputFactory.

**Comment by rfscholte** [comment]
Using Text for these whitespace blocks felt weird but seems to be the only solution without introducing a new object.

**Comment by rolfl** [comment]
OK, so whitespace is clearly allowed in the prolof (and post-root-element) areas of the XML. I agree that Text elements are the logical choice for representing that whitespace, short of introducing a new content type. Text it is.

The test cases using woodstox are causing me headaches. I can run the code standalone, and watch it work, but I can't include your testcase without either it breaking, or other things breaking. The test suites are run using Java 1.7 in ant, I think there's a conflict when attempting to load the woodstox class using the custom class loader:

```
Provider for javax.xml.stream.XMLInputFactory cannot be found

javax.xml.stream.FactoryConfigurationError: Provider for javax.xml.stream.XMLInputFactory cannot be found
at javax.xml.stream.XMLInputFactory.newInstance(XMLInputFactory.java:211)
at org.jdom2.test.cases.input.TestStAXStreamBuilder.checkStAX(TestStAXStreamBuilder.java:147)
at org.jdom2.test.cases.input.TestStAXStreamBuilder.testReportPrologWsDocument(TestStAXStreamBuilder.java:129)

```

My feeling is that this is a very heavyweight test, and can be reduced to testing just the raw Document.addContent method directly with whitespasce. This means neither the full test case nor the StAX and woodstox jar need to be included in the patch set.

I am working on refactoring the commit sequence a bit to not have those Jars in the commit sequence at all.

I can then end up with a simpler problem to look at - just whether the whitespace in Document level content is OK, and what impacts that has on things like the outputter, etc.

For what it;'s worth, adding whitespace to the document level content is the most significant change in the 2.0.x stream as far as I can tell.

**Comment by rolfl** [comment]
CLosing this pull request. PR #156 was used to merge the content in to a separate branch, and in to jdom 2.0.x that way,

**Comment by rfscholte** [comment]
Cool, thanks!

**Pull request closed without merging**